src/diffusers/models/cross_attention.py

```diff
@@ -151,6 +155,16 @@ def set_attention_slice(self, slice_size):
         self.set_processor(processor)
 
     def set_processor(self, processor: "AttnProcessor"):
+        # if current processor is in `self._modules` and if passed `processor` is not, we need to
+        # pop `processor` from `self._modules`
+        if (
+            hasattr(self, "processor")
+            and isinstance(self.processor, torch.nn.Module)
+            and not isinstance(processor, torch.nn.Module)
+        ):
+            logger.info(f"You are removing possibly trained weights of {self.processor} with {processor}")
+            self._modules.pop("processor")
```

Comment:
Looks good ! But what happens when processor is a module ? Do we just overwrite it ? Would be nice to add more comments here, found it a bit hard to understand at first glance.